index.js

```diff
@@ -1460,13 +1460,74 @@ module.exports = function (log, indexesPath) {
     )
   }
 
+  function reindex(offset, cb) {
+    let seq = 0
+    if (offset === 0) {
+      seq = 0
+      offset = -1
+    } else if (offset !== -1) {
+      const { tarr } = indexes['seq']
+      let prevOffset = 0
+      for (const len = tarr.length; seq < len; ++seq)
+        if (tarr[seq] === offset) {
+          offset = prevOffset
```

Comment:
Perhaps it would also help for code readability that the function argument `offset` is not mutated. This code block is reading from `offset` but also reading from `offset`. It sort of changes the semantics of the variable. As a function argument, this is the `firstInvalidOffset`, while the calculation is discovering the `lastValidOffset`. Maybe you can find better names, but these are quick suggestions.